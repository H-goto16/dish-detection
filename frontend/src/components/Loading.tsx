import { View } from 'react-native';
import { ThemedView } from './ThemedView';

const Loading = () => {
  return (
    <ThemedView className='absolute inset-0 items-center justify-center bg-transparent'>
      <View className='h-8 w-8 animate-spin rounded-full border-4 border-blue-500 border-t-transparent' />
    </ThemedView>
  );
};

export default Loading;
