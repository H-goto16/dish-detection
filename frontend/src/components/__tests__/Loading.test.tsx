import type { ReactNode } from 'react';
import renderer from 'react-test-renderer';
import Loading from '../Loading';

// Mock ThemedView component
jest.mock('../ThemedView', () => ({
  ThemedView: ({
    children,
    ...props
  }: { children?: ReactNode; [key: string]: unknown }) => {
    const { View } = require('react-native');
    return <View {...props}>{children}</View>;
  },
}));

describe('Loading', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const component = renderer.create(<Loading />);
    expect(component).toBeTruthy();
  });

  it('renders with correct structure', () => {
    const tree = renderer.create(<Loading />).toJSON();

    // Basic structure validation
    expect(tree).toBeTruthy();
    if (tree && !Array.isArray(tree)) {
      expect(tree.type).toBe('View');
      expect(tree.children).toHaveLength(1);
    }
  });
});
