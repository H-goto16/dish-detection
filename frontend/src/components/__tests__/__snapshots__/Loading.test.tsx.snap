// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<View
  className="absolute inset-0 items-center justify-center bg-transparent"
>
  <View
    className="h-8 w-8 animate-spin rounded-full border-4 border-blue-500 border-t-transparent"
  />
</View>
`;
